### PR TITLE
LibFileSystem: Fix Windows build

### PR DESCRIPTION
**Pull request by shlyakpavel**
This reverts part of  b3c253e50fe630bdf6907fd0641c58975cbcebc5

### CODE REVIEW

File: Libraries/LibFileSystem/FileSystem.cpp [hunk 1]
```diff
@@ -10,6 +10,8 @@
 #include <LibCore/DirIterator.h>
 #include <LibCore/System.h>
 #include <LibFileSystem/FileSystem.h>
+// Required for Windows build
+#include <dirent.h>
```
Comment (gmta) [inline Libraries/LibFileSystem/FileSystem.cpp:14]: Wouldn't it be even more clear that it's Windows-specific if we include it if `AK_OS_WINDOWS` is defined? See two lines below this

Comment (shlyakpavel) [reply Libraries/LibFileSystem/FileSystem.cpp:14]: I wanted to do that first, but when I supposed that this might break Swift bindings

Comment (gmta) [reply Libraries/LibFileSystem/FileSystem.cpp]: I meant including it as an `#elif defined` in the existing block, similar to the other platforms.

Comment (shlyakpavel) [reply Libraries/LibFileSystem/FileSystem.cpp]: Like this?

**Pull request merged**